Common/include/containers/C2DContainer.hpp

```diff
@@ -621,3 +623,40 @@ using su2activematrix = su2matrix<su2double>;
 
 using su2passivevector = su2vector<passivedouble>;
 using su2passivematrix = su2matrix<passivedouble>;
+
+/*--- Functions to allocate 2D/3D data. ---*/
+
+namespace container_helpers {
+
+/*!
+ * \brief Allocate a vector of varying-size vectors and initialize with some value.
+ * \param[in] M - the first index is >=0 and < M
+ * \param[in] N - the second index is >=0 and < N[first index]
+ * \param[in,out] X - the vector of vectors
+ * \param[in] val - the value for initialization, default is 0
+ * \tparam T - type of an element
+ */
+template<typename T>
+inline void Alloc2D(unsigned long M, const std::vector<unsigned long>& N, std::vector<std::vector<T> >& X, T val = 0) {
```

Comment:
You could make the entire `std::vector<unsigned long>` also a template parameter, so that we can pass other types.
Potentially we retrieve "M" from the size of "N", same for Alloc3D